src/components/timeline/ActiveTabGlobalTrack.js

```diff
@@ -0,0 +1,165 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import classNames from 'classnames';
+import { selectActiveTabTrack } from '../../actions/profile-view';
+import {
+  getSelectedThreadIndex,
+  getSelectedTab,
+} from '../../selectors/url-state';
+import explicitConnect from '../../utils/connect';
+import { getActiveTabGlobalTracks } from '../../selectors/profile';
+import './Track.css';
+import TimelineTrackThread from './TrackThread';
+import TimelineTrackScreenshots from './TrackScreenshots';
+import { assertExhaustiveCheck } from '../../utils/flow';
+
+import type { TabSlug } from '../../app-logic/tabs-handling';
+import type { GlobalTrackReference } from '../../types/actions';
+import type {
+  TrackIndex,
+  ActiveTabGlobalTrack,
+  InitialSelectedTrackReference,
+} from '../../types/profile-derived';
+import type { ConnectedProps } from '../../utils/connect';
+
+type OwnProps = {|
+  +trackReference: GlobalTrackReference,
+  +trackIndex: TrackIndex,
+  +style?: Object /* This is used by Reorderable */,
+  +setInitialSelected: (el: InitialSelectedTrackReference) => void,
+|};
+
+type StateProps = {|
+  +globalTrack: ActiveTabGlobalTrack,
+  +isSelected: boolean,
+  +selectedTab: TabSlug,
+|};
+
+type DispatchProps = {|
+  +selectActiveTabTrack: typeof selectActiveTabTrack,
+|};
+
+type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
+
+class ActiveTabGlobalTrackComponent extends PureComponent<Props> {
+  _container: HTMLElement | null = null;
+  _isInitialSelectedPane: boolean | null = null;
+  _selectCurrentTrack = () => {
+    const { selectActiveTabTrack, trackReference } = this.props;
+    selectActiveTabTrack(trackReference);
+  };
+
+  renderTrack() {
+    const { globalTrack } = this.props;
+    switch (globalTrack.type) {
+      case 'tab': {
+        const { threadIndex } = globalTrack;
+        return (
+          <TimelineTrackThread
+            threadIndex={threadIndex}
+            showMemoryMarkers={false}
+          />
+        );
+      }
+      case 'screenshots': {
+        const { threadIndex, id } = globalTrack;
+        return (
+          <TimelineTrackScreenshots threadIndex={threadIndex} windowId={id} />
+        );
+      }
+      default:
+        console.error(
+          'Unhandled active tab globalTrack type',
+          (globalTrack: empty)
+        );
+        return null;
+    }
+  }
+
+  _takeContainerRef = (el: HTMLElement | null) => {
+    const { isSelected } = this.props;
+    this._container = el;
+
+    if (isSelected) {
+      this.setIsInitialSelectedPane(true);
+    }
+  };
+
+  setIsInitialSelectedPane = (value: boolean) => {
+    this._isInitialSelectedPane = value;
+  };
+
+  componentDidMount() {
+    if (this._isInitialSelectedPane && this._container !== null) {
+      this.props.setInitialSelected(this._container);
```

Comment:
It would be good to comment what's going on here with the initial selected. I believe this code was copied/pasted, and it's handling the scrolling of the initial track into view.